src/main/java/seedu/address/logic/parser/ClientNoteAddCommandParser.java

```diff
@@ -25,20 +35,26 @@
     public ClientNoteAddCommand parse(String userInput) throws ParseException {
         requireNonNull(userInput);
         ArgumentMultimap argMultimap =
-                ArgumentTokenizer.tokenize(userInput, /* PREFIX_TAG,*/ PREFIX_NOTE);
+                ArgumentTokenizer.tokenize(userInput, PREFIX_TAG, PREFIX_NOTE);
         if (!arePrefixesPresent(argMultimap, PREFIX_NOTE) || argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, ClientNoteAddCommand.MESSAGE_USAGE));
         }
         Index index;
         try {
-            index = ParserUtil.parseIndex(argMultimap.getPreamble());
+            String[] preambleWords = argMultimap.getPreamble().split(" "); // todo: ask if there's a better way?
+            index = ParserUtil.parseIndex(preambleWords[preambleWords.length - 1]);
```

Comment:
In `AddressBookParser`, the arguments are passed in as `return new ClientNoteAddCommandParser(tagNoteMap).parse(arguments);`. Shouldn't the preamble consist solely of the integer without the command word of `client note add`?

Inside `ClientEditCommandParser`, the index is parsed as such `index = ParserUtil.parseIndex(argMultimap.getPreamble());`

Reply:
Done